src/app/core/guards/auth-guard/auth.guard.ts

```diff
@@ -24,16 +25,19 @@ export class AuthGuard implements CanActivate {
         route: ActivatedRouteSnapshot,
         state: RouterStateSnapshot
     ): Observable<boolean> | Promise<boolean> | boolean {
-        return this.isAuthenticated.pipe(
-            tap((value) => {
-                if (!value) {
-                    this.router.navigate(["login"], {
-                        queryParams: {
-                            loginAgain: true
-                        }
-                    });
-                    this.authService.clearLocalStorage();
+        return this.storeFacadeService.getToken().pipe(
+            take(1),
+            switchMap((token: string) => {

```

Comment:
`map(token => !!token),`
`tap(token => { if (!token) { this.router.navigate(["login"], { queryParams: { loginAgain: true } } ); } })`

Reply:
wow, it is much nicer approach. thank you!